dokka-runners/dokka-gradle-plugin/src/test/kotlin/DokkaPluginTest.kt

```diff
@@ -115,5 +120,74 @@ class DokkaPluginTest : FunSpec({
                 }
             }
         }
+
+        context("DokkaPublications") {
+
+            context("dokka dsl provides convention values for suppressInheritedMembers and suppressObviousFunctions") {
+                val project = ProjectBuilder.builder().build()
+                    .enableV2Plugin()
+                    .also { it.plugins.apply("org.jetbrains.dokka") }
+
+                val dokka = project.extensions.getByType<DokkaExtension>()
+
+                checkAll(
+                    Exhaustive.boolean(),
+                    Exhaustive.boolean(),
+                ) { suppressInheritedMembersValue, suppressObviousFunctionsValue ->
+
+                    @Suppress("DEPRECATION")
+                    dokka.suppressInheritedMembers.set(suppressInheritedMembersValue)
+                    @Suppress("DEPRECATION")
+                    dokka.suppressObviousFunctions.set(suppressObviousFunctionsValue)
+
+                    val publication =
+                        dokka.dokkaPublications.create("TestPublication_${randomSource().random.nextUInt()}")
+
+                    withClue("suppressInheritedMembers should be $suppressInheritedMembersValue") {
+                        publication.suppressInheritedMembers.orNull shouldBe suppressInheritedMembersValue
+                    }
+                    withClue("suppressObviousFunctions should be $suppressObviousFunctionsValue") {
+                        publication.suppressObviousFunctions.orNull shouldBe suppressObviousFunctionsValue
+                    }
+                }
+            }
+
+            context("dokka dsl convention values for suppressInheritedMembers and suppressObviousFunctions can be overridden") {
+                val project = ProjectBuilder.builder().build()
+                    .enableV2Plugin()
+                    .also { it.plugins.apply("org.jetbrains.dokka") }
+
+                val dokka = project.extensions.getByType<DokkaExtension>()
+
+                checkAll(
```

Comment:
nit: I feel this test would be more expressive if it just specified opposing suppression settings for the extension and the publication. Otherwise half of these test cases don't test that the value is actually overriden.

Reply:
Writing out all the different options is quite verbose and error prone, which is why I really like the property-based tests.

Only testing opposing values is a good idea! I've implemented that.